### ISSUE TITLE
Elixir 1.7 and OTP 21 release won't build

### ISSUE THREAD

**Issue by Teifion** [bitwalker/distillery#485]
### Description of issue
I've got an umbrella application with a Phoenix app inside of it. I've upgraded to Elixir 1.7 and am now getting an error trying to build a release.

### Steps to reproduce
`mix release`

I've tried removing and re-initialising the rel directory but get the same result. I would assume it is directly related to Elixir 1.7.

### Log output
<summary>
  <details>

```
Erlang/OTP 21 [erts-10.0] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]
Interactive Elixir (1.7.0) - press Ctrl+C to exit (type h() ENTER for help)


** (Mix.Releases.Config.LoadError) could not load release config rel/config.exs
    ** (UndefinedFunctionError) function Mix.Config.Agent.start_link/0 is undefined (module Mix.Config.Agent is not available)
    Mix.Config.Agent.start_link()
    (stdlib) erl_eval.erl:677: :erl_eval.do_apply/6
    (stdlib) erl_eval.erl:446: :erl_eval.expr/5
    (stdlib) erl_eval.erl:126: :erl_eval.exprs/5
    (elixir) lib/code.ex:232: Code.eval_string/3
    (distillery) lib/mix/lib/releases/config/config.ex:281: Mix.Releases.Config.read_string!/1
    (distillery) lib/mix/lib/releases/config/config.ex:302: Mix.Releases.Config.read!/1
    (distillery) lib/mix/lib/releases/config/config.ex:54: Mix.Releases.Config.get/1
    (distillery) lib/distillery/tasks/release.ex:77: Mix.Tasks.Release.run/1
    (mix) lib/mix/task.ex:316: Mix.Task.run_task/3
    (mix) lib/mix/cli.ex:79: Mix.CLI.run_task/2
    (elixir) lib/code.ex:767: Code.require_file/2
```

  </details>
</summary>

**Comment by stef-otpyrc** [comment]
You can't use any Mix.Config command in your config.exs files.

**Comment by stef-otpyrc** [comment]
Have you try to remove the Mix.Config command in your config.exs files.

**Comment by Teifion** [comment]
I've not got a Mix.Config command in the `rel/config.exs` file (I assume this is the config file you are referring to).

Here is the relevant content from the `rel/config.exs`

```
# Import all plugins from `rel/plugins`
# They can then be used by adding `plugin MyPlugin` to
# either an environment, or release definition, where
# `MyPlugin` is the name of the plugin module.
Path.join(["rel", "plugins", "*.exs"])
|> Path.wildcard()
|> Enum.map(&Code.eval_file(&1))

use Mix.Releases.Config,
    # This sets the default release built by `mix release`
    default_release: :default,
    # This sets the default environment used by `mix release`
    default_environment: Mix.env()
```

**Comment by stef-otpyrc** [comment]
Which version of distillery you are using?
Are you trying to use the Mix Config Provider?
I am referring to the config.exs and prod.exs files in the config folder of your app(s).

**Comment by Teifion** [comment]
Distliery is version 1.5, I also ran `mix deps.get --update-all` to ensure there wasn't an update.

I'm not sure what you mean by the Mix Config Provider, this is the default `rel/config.exs` as generated by distillery (which I regenerated to ensure there wasn't an update to it since I last used it).

My application config files do make use of `use Mix.Config`. I thought you needed to have that as part of a config file.
